### ISSUE TITLE
The result of Internvl-based MM-EUREKA can not be reproduced

### ISSUE THREAD

**Issue by jyC23333** [ModalMinds/MM-EUREKA#35]
# Experiments
Hi, recently I am trying to reproduce MM-EUREKA based on internvl-8B-Instruct. However, I can't reproduce the results.

1. I download the dataset from [huggingface](https://huggingface.co/datasets/FanqingM/MM-Eureka-Dataset).
2. I run a one-node training script with 8xA800.
**(This script can be found here, but removed recently from the repo.  Don't know why)**
```shell
set -x

export RAY_MASTER_PORT=6379
export RAY_DASHBOARD_PORT=8265
export NODE_RANK=0

OUTPUT_DIR='/vepfs-d-data/q-shengtai/changjingyun/projects/MM-EUREKA/results/test_1node_lr3e8'

export REWARD_LOG_PATH="${OUTPUT_DIR}/reward.log"
export WORKING_DIR=$PWD

if [ ! -d "$OUTPUT_DIR" ]; then
  mkdir -p "$OUTPUT_DIR"
fi

if [ "$NODE_RANK" -eq 0 ]; then
    ray start --head  --port=$RAY_MASTER_PORT --dashboard-host=0.0.0.0 --dashboard-port=$RAY_DASHBOARD_PORT --num-gpus 8
else
    sleep 1
    ray start --address="$MASTER_ADDR:$RAY_MASTER_PORT" --num-gpus 8 --block
fi

sleep 1


if [ "$NODE_RANK" -eq 0 ]; then
  RAY_ADDRESS="http://127.0.0.1:$RAY_DASHBOARD_PORT" ray job submit \
  --working-dir $WORKING_DIR \
  --runtime-env examples/runtime_env.yaml \
  -- python3 -m openrlhf.cli.train_ppo_ray \
  --actor_num_nodes 1 \
  --actor_num_gpus_per_node 8 \
  --vllm_num_engines 8 \
  --vllm_tensor_parallel_size 1 \
  --vllm_gpu_memory_utilization 0.7 \
  --vllm_enable_sleep \
  --colocate_all_models \
  --pretrain ./vlm_origin_weights/internvl2_5_8B \
  --remote_rm_url examples/scripts/reward_func_internvl_instruct.py \
  --save_path ${OUTPUT_DIR} \
  --micro_train_batch_size 2 \
  --train_batch_size 64 \
  --micro_rollout_batch_size 4 \
  --rollout_batch_size 128 \
  --temperature 1.0 \
  --n_samples_per_prompt 8 \
  --max_epochs 1 \
  --max_samples 100000 \
  --num_episodes 1 \
  --prompt_max_len 2048 \
  --generate_max_len 4096 \
  --zero_stage 1 \
  --bf16 \
  --actor_learning_rate 3e-8 \
  --advantage_estimator rloo \
  --init_kl_coef 0.0 \
  --prompt_data ./MM-Eureka-Dataset/dataset.jsonl \
  --apply_chat_template \
  --disable_fast_tokenizer \
  --normalize_reward \
  --adam_offload \
  --flash_attn \
  --gradient_checkpointing \
  --save_steps 50 \
  --use_tensorboard "${OUTPUT_DIR}/tensorboard" \
  --ckpt_path "${OUTPUT_DIR}/ckpt" \
  --max_ckpt_num 1000 \
  --save_hf_ckpt \
  --load_checkpoint | tee ${OUTPUT_DIR}/training.log
fi

# --runtime-env-json='{"setup_commands": ["pip install openrlhf[vllm]"]}' [Install deps]
# --ref_reward_offload [Offload to CPU]
# --remote_rm_url http://localhost:5000/get_reward
``` 
3. The curves of the reward and response length seemed abnormal.
  - acc_reward
![Image](https://github.com/user-attachments/assets/ebc91453-c900-46ad-b4bb-cc7877e7af52)
  - format_reward
![Image](https://github.com/user-attachments/assets/bab8f5c0-7cc8-451c-9e89-4262a36273fc)
  - response length
![Image](https://github.com/user-attachments/assets/9688d71c-5660-4c0b-b1f0-478b87ca03be)
4. and then I eval the model on `K12`, but I can only reach 0.10(acc).

# Other
- I found my curve is similiar to the w/o data filting curve in [official Technical Report](https://arxiv.org/pdf/2503.07365):
![Image](https://github.com/user-attachments/assets/37df9627-7680-4955-8b9c-bfec8ba31260)

**Comment by FanqingM** [comment]
We upload the tensorboard and training.log at https://github.com/ModalMinds/MM-EUREKA/tree/internvl/log
